hoomd/md/minimize/fire.py

```diff
@@ -0,0 +1,206 @@
+from hoomd.logging import log
+from hoomd.md import _md
+from hoomd.md.integrate import _DynamicIntegrator, _preprocess_aniso
+
+
+def _raise_if_nonpositive(value):
+    if value <= 0:
+        raise ValueError("This value cannot be negative")
+    return value
+
+
+class FIRE(_DynamicIntegrator):
+    """Energy Minimizer (FIRE).
+
+    Args:
+        dt (float):
+            This is the maximum step size the minimizer is permitted to use.
+            Consider the stability of the system when setting.
+            math:`[\\mathrm{time}]`
+        min_steps_adapt (int):
+            Number of steps energy change is negative before allowing
+            :math:`\\alpha` and :math:`\\delta t` to adapt.
+        finc_dt (float):
+            Factor to increase :math:`\\delta t` by
+        fdec_dt (float):
+            Factor to decrease :math:`\\delta t` by
+        alpha_start (float):
+            Initial (and maximum) :math:`\\alpha`
+        fdec_alpha (float):
+            Factor to decrease :math:`\\alpha t` by
+        force_tol (float):
+            force convergence criteria
+            math:`[\\mathrm{force} / \\mathrm{mass}]`
+        angmom_tol (float):
+            angular momentum convergence criteria
+            math:`[\\mathrm{energy} * \\mathrm{time}]`
+        energy_tol (float):
+            energy convergence criteria math:`[\\mathrm{energy}]`
+        min_steps_conv (int):
+            A minimum number of attempts before convergence criteria are
+            considered
+        aniso (bool):
+            Whether to integrate rotational degrees of freedom (bool), default
+            None (autodetect).
+
+    `minimize.FIRE` uses the Fast Inertial Relaxation Engine (FIRE) algorithm
+    to minimize the energy for a group of particles while keeping all other
+    particles fixed.  This method is published in `Bitzek, et. al., PRL, 2006
+    <http://dx.doi.org/10.1103/PhysRevLett.97.170201>`_.
+
+    At each time step, :math:`\\delta t`, the algorithm uses the NVE Integrator
+    to generate a x, v, and F, and then adjusts v according to
+
+    .. math::
+
+        \\vec{v} = (1-\\alpha)\\vec{v} + \\alpha \\hat{F}|\\vec{v}|
+
+    where :math:`\\alpha` and :math:`\\delta t` are dynamically adaptive
+    quantities.  While a current search has been lowering the energy of system
+    for more than :math:`N_{min}` steps, :math:`\\alpha` is decreased by
+    :math:`\\alpha \\rightarrow \\alpha fdec_{\\alpha}` and :math:`\\delta t`
+    is increased by :math:`\\delta t \\rightarrow max(\\delta t \\cdot
+    finc_{dt}, \\ \\delta t_{max})`. If the energy of the system increases (or
+    stays the same), the velocity of the particles is set to 0, :math:`\\alpha
+    \\rightarrow \\ \\alpha_{start}` and :math:`\\delta t \\rightarrow \\delta t
+    \\cdot fdec_{\\alpha}`. Convergence is determined by both the force per
+    particle and the change in energy per particle dropping below `angmom_tol`
+    and `energy_tol`, respectively or
```

Comment:
```suggestion
    \\cdot fdec_{\\alpha}`. The method converges when the force per
    particle is below `force_tol`, the angular momentum is below `angmom_tol`
    and the change in potential energy from one step to the next is below 
    `energy_tol`:
```
You also need to update the equation.

Reply:
Which part of which equation needs to be updated?

Reply:
The equation just below this comment:
```
        \\frac{\\sum |F|}{N*\\sqrt{N_{dof}}} < \\mathrm{\\text{angmom_tol}}
        \\;\\; and \\;\\ \\Delta \\frac{\\sum|E|}{N} <
        \\mathrm{\\text{energy_tol}}
```
It incorrectly compares forces to `angmom_tol` and only lists 2 of the 3 conditions.

Reply:
Also, should this suggestion say "the angular momentum per particle is below `angmom_tol` and the change ..."?